packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -168,78 +169,82 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, IERC900
     }
 
     /**
-    * @notice Activate `_amount == 0 ? 'all available tokens' : @tokenAmount(self.token(), _amount)` for the next term
-    * @param _amount Amount of guardian tokens to be activated for the next term
+    * @notice Stake `@tokenAmount(self.token(), _amount)` for `_guardian`
+    * @param _guardian Address of the guardian to stake tokens to
+    * @param _amount Amount of tokens to be staked
+    * @param _data Optional data is never used by this function, only logged
     */
-    function activate(uint256 _amount) external {
-        _activateTokens(msg.sender, _amount, msg.sender);
+    function stake(address _guardian, uint256 _amount, bytes calldata _data) external {
```

Comment:
Should this include an `authenticate(_guardian)`?

Reply:
Stake is the only one that is not harmful, and I think it should be fine we allow other people to stake on behalf of some guardians. It would be like signaling, they would be basically gifting money

Reply:
Oh this is smart. This clarifies how `stakeAndActivate()` work as well then :).